modules/core/nuxeo-core-storage-mem/src/test/java/org/nuxeo/ecm/core/storage/mem/MockRepositoryManager.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * (C) Copyright 2021 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Anahide Tchertchian
+ */
+package org.nuxeo.ecm.core.storage.mem;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.nuxeo.ecm.core.api.repository.Repository;
+import org.nuxeo.ecm.core.api.repository.RepositoryManager;
+import org.nuxeo.runtime.model.ComponentContext;
+import org.nuxeo.runtime.model.DefaultComponent;
+
+/**
+ * Mock repo manager for tests.
+ * <p>
+ * Implemented methods prevent further setup of repository: tests will only check addition of the repository through the
+ * {@link #addRepository(Repository)} api.
+ *
+ * @since 11.5
+ */
+public class MockRepositoryManager extends DefaultComponent implements RepositoryManager {
```

Comment:
Not all of them are implemented, the goal was to check that `#addRepository` was properly called (to show that mem and mongodb contributions were properly registered up to RepositoryManager).
Some methods like `#getRepositoryNames` returns an empty list to prevent further setup of the repository, which is indeed an issue in this context.

Overall, these are "unit" tests of the components registration: to setup a full repo, using the `CoreFeature` would be better but it's not possible here (maven cycle dep) so I went for a pseudo-mock implementation (which helped checking the corresponding registry behavior was alright, still).
To simplify all this, we'd need to merge some core modules (as @efge mentioned again recently).